packages/common/src/apiutil/functions.ts

```diff
@@ -184,23 +185,29 @@ export function getRepositoryData(request: Request ): RepositoryData | Parameter
     }
 }
 
+let index = 1
 /**
- * Catch-all wrapper function to handle exceptions for any api call
+ * Catch-all wrapper function to handle exceptions for any api call.
+ * And put the request function in the request queue.
  * @param func
  */
 export function runWithTry(func: (request: Request, response: Response) => void): (request: Request, response: Response) => void {
     return async function (request: Request, response: Response): Promise<void> {
-        try {
-            await func(request, response)
-        } catch (e) {
-            const error = asError(e)
-            requestLogger.error(`Exception while serving request for ${request.url}: ${error.message}`)
-            requestLogger.error(error)
-            lionwebResponse(response, HttpServerErrors.InternalServerError, {
-                success: false,
-                messages: [{ kind: error.name, message: `Exception while serving request for ${request.url}: ${error.message}` }]
-            })
+        const myIndex = index++
```

Comment:
Stupid question: are we sure this operation is atomic? Could we ever have two of these operations running in parallel?

Reply:
This is **the smartest question** to ask, as everything depends on the operations always running sequentially.
This `runWithTry` function returns a puts the actual request function on the request queue.

The request queue then makes sure all operations are executed sequentially and never in parallel. This behaviour is pretty hard to to implement and almost impossible to test. All test work properly, but if there is a tiny loophole where things might go wrong, you cannot write a test to check this. It is also no easy to prove.

So I just read the code in the RequestQueueclass step by step and think it is working ok now, without loopholes.